runtime/account/src/signer.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright 2019-2020 PureStake Inc.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+//! The Ethereum Signature implementation.
+//!
+//! It includes the Verify and IdentifyAccount traits for the AccountId20
+
+use sp_core::{H160, H256, ecdsa, ed25519, sr25519, RuntimeDebug};
+use codec::{Decode, Encode, Input};
+use sha3::{Digest, Keccak256};
+
+#[cfg(feature = "std")]
+pub use serde::{Serialize, Deserialize, de::DeserializeOwned};
+
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Eq, PartialEq, Clone, Encode, Decode, sp_core::RuntimeDebug)]
+pub struct EthereumSignature(ecdsa::Signature);
+
+impl From<ecdsa::Signature> for EthereumSignature {
+	fn from(x: ecdsa::Signature) -> Self {
+		EthereumSignature(x)
+	}
+}
+
+#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+#[derive(Eq, PartialEq, Clone, Encode, Decode, RuntimeDebug)]
+pub enum MultiSignature {
+	Ed25519(ed25519::Signature),
+	Sr25519(sr25519::Signature),
+	Ecdsa(EthereumSignature),
+}
```

Comment:
Do we have any desire to use ed25519 or sr25519? If not, this should definitely go.

Reply:
Do you think of any usecase to support those so far?
Otherwise yes, iit should go but we don't have a way to support it on PolkadotJs.